packages/components/src/internal/components/samples/SampleAssayDetail.tsx

```diff
@@ -159,50 +270,173 @@ const SampleAssayDetailBodyImpl: FC<SampleAssayDetailBodyProps & InjectedQueryMo
 const SampleAssayDetailBody = withQueryModels<SampleAssayDetailBodyProps>(SampleAssayDetailBodyImpl);
 
 const SampleAssayDetailImpl: FC<Props & InjectedAssayModel> = props => {
-    const { assayModel, sampleId, sampleModel } = props;
+    const {
+        assayModel,
+        sampleId,
+        sampleModel,
+        sampleAliquotType,
+        showAliquotViewSelector,
+        sourceSampleIds,
+        sourceAliquotIds,
+        sourceId,
+    } = props;
     const loadingDefinitions = isLoading(assayModel.definitionsLoadingState);
 
+    const [aliquotIds, setAliquotIds] = useState(undefined);
+    const [activeSampleAliquotType, setActiveSampleAliquotType] = useState(
+        sampleAliquotType ?? ALIQUOT_FILTER_MODE.all
+    );
+
+    const isSourceSampleAssayGrid = useMemo(() => {
+        return sampleId == null && sourceSampleIds != null;
+    }, [sampleId, sourceSampleIds]);
+
+    useEffect(() => {
+        if (!showAliquotViewSelector || !sampleId) return;
+
+        getSampleAliquots(sampleId)
+            .then(aliquots => {
+                setAliquotIds(aliquots);
+            })
+            .catch(reason => {
+                createNotification({
+                    alertClass: 'danger',
+                    message: 'Unable to load sample aliquots',
```

Comment:
Do we have any suggestions for what they might try to fix this?  (e.g., "Make sure the samples are still valid."??).  Doesn't have to have anything added, but if we know likely candidates for what could go wrong, it would be good to help the user to troubleshoot.